src/lib.rs

```diff
@@ -369,8 +369,15 @@ impl<W: Write> Archive<W> {
     }
 
     fn append_path2(&self, path: &Path) -> io::Result<()> {
-        let mut file = try!(fs::File::open(path));
-        self.append_file(path, &mut file)
+        let stat = try!(fs::metadata(path));
+        if stat.is_file() {
+            let mut file = try!(fs::File::open(path));
+            self.append_file(path, &mut file)
```

Comment:
It could, but then you've got the header building in two places? By calling `append_path` there's only one codepath for it.
I don't really have strong feelings - want me to move file header building into here?

This could also call the `append_{file,dir}2` functions instead of the public ones.